frontend/server/src/main/java/org/pytorch/serve/wlm/ContinuousBatching.java

```diff
@@ -0,0 +1,138 @@
+     */
+    public void sendResponse(ModelWorkerResponse message) {
+        boolean jobDone = true;
+        // TODO: Handle prediction level code
+        if (message.getCode() == 200) {
+            if (jobs.isEmpty()) {
+                // this is from initial load.
+                return;
+            }
+            for (Predictions prediction : message.getPredictions()) {
+                String jobId = prediction.getRequestId();
+                Job job = jobs.get(jobId);
+
+                if (job == null) {
+                    throw new IllegalStateException(
+                            "Unexpected job in sendResponse() with 200 status code: " + jobId);
+                }
+
+                if (job.getPayload().getClientExpireTS() > System.currentTimeMillis()) {
+                    job.response(
+                            prediction.getResp(),
+                            prediction.getContentType(),
+                            prediction.getStatusCode(),
+                            prediction.getReasonPhrase(),
+                            prediction.getHeaders());
+                } else {
+                    logger.warn(
+                            "Drop response for inference request {} due to client timeout",
+                            job.getPayload().getRequestId());
+                }
+                String streamNext =
+                        prediction
+                                .getHeaders()
+                                .get(org.pytorch.serve.util.messages.RequestInput.TS_STREAM_NEXT);
+                if (streamNext != null && streamNext.equals("false")) {
+                    jobs.remove(job.getJobId());
+                }else if (!job.isOpen()){
+                    jobs.remove(job.getJobId());
+                    logger.info("Connection to client got closed; Removing job: {}",job.getPayload().getRequestId());
+                }
+
+            }
+        } else {
+            for (Map.Entry<String, Job> j : jobs.entrySet()) {
+                if (j.getValue() == null) {
+                    throw new IllegalStateException(
+                            "Unexpected job in sendResponse() with non 200 status code: "
+                                    + j.getKey());
+                }
+                Job job = j.getValue();
+                if (job.getPayload().getClientExpireTS() > System.currentTimeMillis()) {
+                    job.sendError(message.getCode(), message.getMessage());
+                } else {
+                    logger.warn(
+                            "Drop error response for inference request {} due to client timeout",
+                            job.getPayload().getRequestId());
+                }
+            }
+            jobs.clear();
+        }
+    }
+
+    private void pollBatch(String threadName, WorkerState state, int batchSize)
+            throws InterruptedException {
+        if (!model.pollMgmtJob(
```

Comment:
we call pollMgmtJob only if jobs.isEmpty, otherwise. it is possible to mix mgmt job with inference job and break the backend

Reply:
see above